milli/src/error.rs

```diff
@@ -47,6 +47,8 @@ pub enum InternalError {
     IndexingMergingKeys { process: &'static str },
     #[error("{}", HeedError::InvalidDatabaseTyping)]
     InvalidDatabaseTyping,
+    #[error("Could not access the inner of a buf-reader/writer: {0}")]
+    BufIntoInnerError(String),
```

Comment:
I don't know if we can eliminate this variant or simply change this string into an `io::Error`. See my other comment.